src/wallet/api/wallet2_api.h

```diff
@@ -1144,6 +1149,545 @@ struct Wallet
+    /**
+    * brief: isOutputBlackballed -
+    * param: output - [amount, offset] for ringdb index
+    * return: true if succeeded
+    * note: sets status error on fail
+    */
+    virtual bool isOutputBlackballed(const std::pair<std::uint64_t, std::uint64_t> &output) const = 0;
+    // QUESTION : Any suggestion for the two comments below?
+    /**
+    * brief: coldTxAuxImport -
+    * param: ptx -
+    * param: tx_device_aux -
+    * note: sets status error on fail
+    */
+    virtual void coldTxAuxImport(const PendingTransaction &ptx, const std::vector<std::string> &tx_device_aux) const = 0;
+// TODO : wallet2::signed_tx_set & cryptonote::address_parse_info
+    /**
+    * brief: coldSignTx -
+    * param: ptx -
+    * param: exported_txs -
+    * param: dsts_info -
+    * param: tx_device_aux -
+    */
+//    virtual void coldSignTx(const PendingTransaction &ptx, signed_tx_set &exported_txs, std::vector<cryptonote::address_parse_info> &dsts_info, std::vector<std::string> & tx_device_aux) const = 0;
+    // TODO : this already exists, remove after checking
+    /**
+    * brief: useForkRules - check if the wallet uses the rules for hard fork version at current block height - early_blocks
+    * param: version - hard fork version
+    * param: early_blocks -
+    * return: true if fork rules are used
+    * note: sets status error on fail
+    */
+//    virtual bool useForkRules(std::uint8_t version, std::int64_t early_blocks) const = 0;
+// TODO : wallet2::transfer_details
+    /**
+    * brief: getTransferDetails -
+    * param: idx - index in transfer storage
+    * return: transfer details for given index
+    * note: sets status error on fail
+    */
+//    virtual const wallet2::transfer_details &getTransferDetails(std::size_t idx) const = 0;
+    /**
+    * brief: discardUnmixableOutputs - freeze all unmixable outputs
+    * note: sets status error on fail
+    */
+    virtual void discardUnmixableOutputs() = 0;
+    /**
+    * brief: setTxKey - set the transaction key (r) for a given <txid> in case the tx was made by some other device or 3rd party wallet
+    * param: txid -
+    * param: tx_key - secret transaction key r
+    * param: additional_tx_keys -
+    * param: single_destination_subaddress -
+    * note: sets status error on fail
+    */
+    // QUESTION : There were some cases already where I wished to have boost::optional in here. I have no explicit example, but in general e.g. for functions that return a bool, but have to return early on error, it would be cleaner to return boost::none instead of true/false, because the API user could forget to check the error status and may use an incorrect value. Similar for uint*_t, where we can't use e.g. -1 on error.
+    //            If we add access to boost::optional revisit all new functions to see which would benefit from it.
```

Comment:
Hmm, interesting question, but I think it would really help to see some good examples / candidates before really discussing this.

In my experience devs rarely really forget to check error status. What often happens instead is that they simply judge an error to be very, very unlikely and opt for neglecting to test. And if a dev does not "feel like testing", you can design your call whatever way you want, they will find a way to just throw away any result or just use something that you give back without testing it first for null, boost::none or whatever.

IMHO if it's clear that you do give back a result from the design of a method the design job is mostly already done.